stock_location_zone/models/stock_location.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 Sylvain Van Hoof <svh@sylvainvh.be>
+# Copyright 2018-2019 Jacques-Etienne Baudoux (BCIM sprl) <je@bcim.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+
+
+class StockLocation(models.Model):
+    _inherit = 'stock.location'
+
+    # FIXME: add in selection: shuttle, tray (module vertical lift)
+    kind = fields.Selection([
+        ('zone', 'Picking Zone'),
+        ('area', 'Area'),
+        ('bin', 'Bin'),
+        ],
+        string='Kind')
+
+    picking_zone_id = fields.Many2one(
+        'picking.zone',
+        string='Picking zone')
+
+    picking_type_id = fields.Many2one(
+        related='picking_zone_id.pick_type_id',
+        help="Picking type for operations from this location",
+        oldname='barcode_picking_type_id')
+
+    area = fields.Char(
+        'Area',
+        compute='_compute_area', store=True,
+        oldname='zone')
+
+    @api.depends('name', 'kind', 'location_id.area')
+    def _compute_area(self):
+        for location in self:
+            if location.kind == 'area':
+                location.area = location.name
+            else:
+                location.area = location.location_id.area
+
+    corridor = fields.Char('Corridor', help="Street")
+    row = fields.Char('Row', help="Side in the street")
+    rack = fields.Char('Rack', oldname='shelf', help="House number")
+    level = fields.Char('Level', help="Height on the shelf")
+    posx = fields.Integer('Box (X)')
+    posy = fields.Integer('Box (Y)')
+    posz = fields.Integer('Box (Z)')
+
+    location_name_format = fields.Char(
+        'Location Name Format',
+        help="Format string that will compute the name of the location. "
+             "Use 'self' to access location object. Example: "
```

Comment:
This is too unsafe, allowing too much python code, we should allow a safe list of keywords

Reply:
Changed object argument to **object.read()

Reply:
It fails with the following error on a new record:

```
 The requested operation cannot be completed due to record rules: Document type: stock.location, Operation: read, Records: <odoo.models.NewId object at 0x7f659f6921c8>, User: 2
```